### PR TITLE
Feature/bma/rr

### PR DESCRIPTION
**Pull request by bmarty**
### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/RoomSyncEphemeralTemporaryStore.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Read RoomSyncEphemeral from a file, or null if there is no file to read
+     */
+    override fun read(roomId: String): RoomSyncEphemeral? {
+        return getFile(roomId)
+                .takeIf { it.exists() }
+                ?.inputStream()
+                ?.use { pos ->
+                    MoshiProvider.providesMoshi().adapter(RoomSyncEphemeral::class.java)
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/RoomSyncEphemeralTemporaryStore.kt:60]: inject moshi?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/ReadReceiptHandler.kt [hunk 2]
```diff
@@ -110,4 +122,13 @@ internal class ReadReceiptHandler @Inject constructor() {
             }
         }
     }
+
+    fun getContentFromInitSync(roomId: String): ReadReceiptContent? {
+        @Suppress("UNCHECKED_CAST")
+        return roomSyncEphemeralTemporaryStore.read(roomId)
+                ?.also { roomSyncEphemeralTemporaryStore.delete(roomId) }
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/ReadReceiptHandler.kt:129]: If something fails during the sync transaction we will loose the initial read receipts

**Review by ganfra — COMMENTED** [review]
2 remarks, otherwise LGTM

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/ReadReceiptHandler.kt (hunk 2 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/ReadReceiptHandler.kt:129]: Yes, good point, I'm not sure how to handle that... Will try to find a solution

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/RoomSyncEphemeralTemporaryStore.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/RoomSyncEphemeralTemporaryStore.kt:60]: Yes, will do

**Review by ganfra — APPROVED** [review]

**Pull request merged**